scrapy/crawler.py

```diff
@@ -135,6 +118,26 @@ def crawl(self, *args: Any, **kwargs: Any) -> Generator[Deferred, Any, None]:
 
         try:
             self.spider = self._create_spider(*args, **kwargs)
+            self.spider.update_settings(self.settings)
```

Comment:
I wonder if update_settings running after `__init__` now could be an issue to some existing code, e.g. code reading settings from `__init__`.

Also, update_settings kind of becomes unnecessary, since spiders could now use crawler.settings from `__init__` instead for the same purpose. Still convenient, I guess.

I am thinking, although we do need setting processing to happen within `crawl`, maybe it could still happen before `spider.__init__`.

And we could expose *args and **kwargs to update_settings, with some backward-compatibility code to not pass them if unexpected, or maybe better, through a new method (of extensions?) that gets executed after setting processing but before spider init, gets access to argument and (unfrozen) settings, and can edit both.